homeassistant/components/husqvarna_automower/coordinator.py

```diff
@@ -35,7 +36,10 @@ async def _async_update_data(self) -> dict[str, MowerAttributes]:
             await self.api.connect()
             self.api.register_data_callback(self.callback)
             self.ws_connected = True
-        return await self.api.get_status()
+        try:
+            return await self.api.get_status()
+        except ApiException as err:
+            raise UpdateFailed(err) from err
```

Comment:
That only catches exceptions while polling. But the websocket still throws Exceptions which aren't catched:
```
2024-02-21 19:14:50.582 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 1173, in _create_direct_connection
    hosts = await asyncio.shield(host_resolved)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 884, in _resolve_host
    addrs = await self._resolver.resolve(host, port, family=self._family)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/resolver.py", line 33, in resolve
    infos = await self._loop.getaddrinfo(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 867, in getaddrinfo
    return await self.run_in_executor(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/vscode/.local/lib/python3.11/site-packages/aioautomower/session.py", line 112, in _receiver
    websocket: ClientWebSocketResponse = await self.auth.websocket_connect()
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/aioautomower/auth.py", line 180, in websocket_connect
    self._ws = await self._websession.ws_connect(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 832, in _ws_connect
    resp = await self.request(
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 578, in _request
    conn = await self._connector.connect(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 544, in connect
    proto = await self._create_connection(req, traces, timeout)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 911, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 1187, in _create_direct_connection
    raise ClientConnectorError(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host ws.openapi.husqvarna.dev:443 ssl:default [Temporary failure in name resolution]
```
How can that be achived?

Reply:
I mentioned this in the first PR. It's easier to handle exceptions if the library would let the user of the library to create the task, and manage it, to receive from the websocket. Then the library user can add an exception handler around that method before creating the task. And the library should raise library specific exceptions instead of aiohttp exceptions.

Reply:
Example:
https://github.com/home-assistant/core/blob/8bf0466151d5a7bebcd81a983c11536b49b9a934/homeassistant/components/matter/__init__.py#L102-L107

https://github.com/home-assistant/core/blob/8bf0466151d5a7bebcd81a983c11536b49b9a934/homeassistant/components/matter/__init__.py#L139-L160